### PR TITLE
EBS UI Reforming

### PR DESCRIPTION
**Pull request by root-beer**
### GOAL
- [x] Accordion UI
- [x] Policy and Schedule Indicator

### CODE REVIEW

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 1]
```diff
@@ -600,6 +622,24 @@ module.exports = Vue.extend({
...
 
+    is_retention_policy_set: function () { return this.ec2.retention_policies[this.volume_selected].enabled },
+    is_snapshot_schedule_set: function () { return this.ec2.snapshot_schedules[this.volume_selected].enabled },
+
+    schedule_indicator_message: function () {
+      var schedule = this.ec2.snapshot_schedules[this.volume_selected];
+      switch (schedule.frequency) {
+        case 'intervals':
+          return t('schedules.label.per_n_hours', { n: schedule.time });
+          break;
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: __[jshint]__
*  Unreachable 'break' after 'return'.

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 2]
```diff
@@ -600,6 +622,24 @@ module.exports = Vue.extend({
...
+        case 'daily':
+          return t('schedules.label.daily', { n: schedule.time });
+          break;
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: __[jshint]__
*  Unreachable 'break' after 'return'.

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 3]
```diff
@@ -600,6 +622,24 @@ module.exports = Vue.extend({
...
+        case 'weekly':
+          return t('schedules.label.weekly', { n: schedule.time, w: t('schedules.day_of_week.' + schedule.day_of_week) });
+          break;
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: __[jshint]__
*  Unreachable 'break' after 'return'.

File: app/models/ec2_instance.rb [hunk 4]
```diff
@@ -147,11 +147,17 @@ def attach_volume(volume_id, device_name)
 
   def retention_policies
     volume_ids = block_device_mappings.map { |e| e.ebs.volume_id }
-    policies = RetentionPolicy.where(resource_id: volume_ids)
-    policies.map { |e|
-      [ e.resource_id,
-        { max_amount: e.max_amount },
+    policies = RetentionPolicy.where(resource_id: volume_ids).map { |policy|
+      [ policy.resource_id,
```
Comment (skyarch-networks) [inline app/models/ec2_instance.rb]: __[rubocop]__
* Style/MultilineArrayBraceLayout: Closing array brace must be on the same line as the last array element when opening brace is on the same line as the first array element.

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://sideci.com/github_repositories/31754199/auto_correct/rubocop/pull_request/269/branch/reform-ebs-ui)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by root-beer** [comment]
Please review and merge PR. @Joeper214 :bow:

**Comment by Joeper214** [comment]
I'll review Tomorrow. :+1:

**Comment by Joeper214** [comment]
I encountered some error during snapshot creation. @root-beer 

error

File: app/views/infrastructures/modal/_modal_snapshot_retention_policy.html.erb [hunk 5]
```diff
@@ -19,14 +19,14 @@
             </div>
           </div>
           <div class="col-md-3">
-            <input class="col-xs-12" :disabled="!editing_policy.enabled" v-model="editing_policy.max_amount" id="editing_policy.max_amount" type="number" min="3">
+            <input class="col-xs-12" :disabled="!editing_policy.enabled" v-model="editing_policy.max_amount" id="editing_policy.max_amount" type="number" min="3" max="999" maxlength="3">
```
Comment (Joeper214) [inline app/views/infrastructures/modal/_modal_snapshot_retention_policy.html.erb]: :+1:

File: app/assets/javascripts/infrastructures/ec2-tabpane.js [hunk 6]
```diff
@@ -601,6 +623,22 @@ module.exports = Vue.extend({
       return '/dev/sd' + String.fromCharCode(suggested_device_letter_code);
     },
 
+    is_valid_amount: function () { return 3 <= this.editing_policy.max_amount && this.editing_policy.max_amount < 1000; },
+    is_retention_policy_set: function () { return this.ec2.retention_policies[this.volume_selected].enabled },
```
Comment (Joeper214) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: Maybe this method sets the error. I think we should check `this.ec2.retention_policies[this.volume_selected].enabled` first if it's null before returning it. 

error

Comment (root-beer) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: Okay. I'll fix now.

Comment (Joeper214) [inline app/assets/javascripts/infrastructures/ec2-tabpane.js]: arigatou. :bow:

**Comment by root-beer** [comment]
Please review and merge PR. @Joeper214 :bow:

**Comment by Joeper214** [comment]
I think we need to get the failed message  when attaching a volume in this line:


[Attach_volume](https://github.com/skyarch-networks/skyhopper/blob/9459c86cb0c0870e15c9e37829657c13cc7bc757/app/assets/javascripts/infrastructures/ec2-tabpane.js#L434)

Because I tried to attach something and there is no error message return.

attach

**Comment by Joeper214** [comment]
I think we need to include this.
 
`.fail(alert_danger());`

**Comment by Joeper214** [comment]
On this following line. 

```javascript
modal.Prompt(t('ec2_instances.set_device_name'), t('ec2_instances.device_name')).done(function (device_name) {
        ec2.attach_volume(volume_id, device_name).done(function (data) {
          modal.Alert(t('infrastructures.infrastructure'), t('ec2_instances.msg.volume_attached', data)).done(self._show_ec2);
        });
      });
```

**Comment by root-beer** [comment]
Thanks! I fixed it.

**Comment by Joeper214** [comment]
LGTM! Thanks.

**Comment by root-beer** [comment]
:bow:

**Pull request merged**